### PR TITLE
Selects a delivery option, estimates shipping and make some UI adjustments

### PR DESCRIPTION
**Pull request by jeffersontuc**
#### Notes

This PR depends on and must be merged after [order-shipping](https://github.com/vtex-apps/order-shipping/pull/2).

#### What problem is this solving?

In order to select a delivery option and to estimate shipping value, the component needs to use `OrderShipping` API. This PR does it. 

It also makes some UI adjustments.

#### How should this be manually tested?

[Workspace](https://ordershipping--vtexgame1.myvtex.com/cart)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] [Linked this PR to a Clubhouse story (if applicable).](https://app.clubhouse.io/vtex/story/18320/implementar-consumer-do-order-shipping-no-shipping-calculator)
- [x] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️| New feature 
_ | Breaking change 
✔️| Technical improvements

### CODE REVIEW

File: react/components/EstimateShipping.tsx [hunk 1]
```diff
@@ -4,39 +4,52 @@ import ShippingResult from './ShippingResult'
...
+    addValidation(
+      selectedAddress
+        ? newAddress(selectedAddress)
+        : newAddress({ country: culture.country })
+    )
   )
 
-  const [showResult, setShowResult] = useState<boolean>(false)
+  const [showResult, setShowResult] = useState<boolean>(
+    selectedAddress && selectedAddress.postalCode ? true : false
```
Comment (darlenedms) [inline react/components/EstimateShipping.tsx:37]: Not a change request, just a comment, I guess you could've used the `!!` notation instead of ternary.

`(!!selectedAddress && selectedAddress.postalCode)`

File: react/components/ShippingInfo.tsx [hunk 2]
```diff
@@ -8,16 +8,16 @@ interface CustomProps {
...
-        <span>
-          <FormattedCurrency value={option.value / 100} />
-        </span>
-      </p>
-      <p className="c-muted-1 mv3">
-        <TranslateEstimate shippingEstimate={option.estimate} />
-      </p>
+    <div className="flex w-100">
+      <div className="flex-auto">
+        <div className="mb3">{option.id}</div>
```
Comment (darlenedms) [inline react/components/ShippingInfo.tsx:13]: Why have you changed from `p` to `div`, I mean, why was that necessary?

**Review by darlenedms — APPROVED** [review]
LGTM

File: react/ShippingCalculator.tsx [hunk 3]
```diff
@@ -14,35 +16,40 @@ defineMessages({
...
 })
 
-type CustomProps = {
-  selectedAddress: Address
-  deliveryOptions: DeliveryOption[]
-  countries: string[]
-}
+interface CustomProps {}
+
+const ShippingCalculator: FunctionComponent<CustomProps> = () => {
```
Comment (marcoskwkm) [inline react/ShippingCalculator.tsx]: Delete the empty interface.
`const ShippingCalculator: FunctionComponent = () => {`

File: react/__mocks__/vtex.address-form.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+} from '@vtex/address-form/lib/validateAddress'
+import InputError from '@vtex/address-form/lib/inputs/DefaultInput/InputError'
+import InputText from '@vtex/address-form/lib/inputs/DefaultInput/InputText'
+import DefaultInput from '@vtex/address-form/lib/inputs/DefaultInput'
+import AddressShapeWithValidation from '@vtex/address-form/lib/propTypes/AddressShapeWithValidation'
+import AddressSummary from '@vtex/address-form/lib/AddressSummary'
+
+// Added react context for injectRules mock
+// eslint-disable-next-line
+import React, { Component } from 'react'
```
Comment (marcoskwkm) [inline react/__mocks__/vtex.address-form.tsx]: Do you need `React` here?

File: react/__mocks__/vtex.address-form.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+function injectRules(component) {
```
Comment (marcoskwkm) [inline react/__mocks__/vtex.address-form.tsx]: `component: Component`

File: react/components/ShippingInfo.tsx (hunk 2 above)
Comment (jeffersontuc) [reply react/components/ShippingInfo.tsx:13]: I think @omninando can answer you.

Comment (jeffersontuc) [reply react/components/ShippingInfo.tsx:13]: Actually, I just remembered it was @augustocb . He told me something about `p` tag semantics.

File: react/__mocks__/vtex.format-currency.tsx [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const FormattedCurrency = (args: any) => {
```
Comment (marcoskwkm) [inline react/__mocks__/vtex.format-currency.tsx]: `export const FormattedCurrency = (_: any) => {`

File: react/__mocks__/vtex.address-form.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+const components = {
+  AddressSummary,
+  AddressContainer,
+  AddressRules,
+  CountrySelector,
+  PostalCodeGetter,
+}
+
+module.exports = { components, countries, inputs, helpers, shapes }
```
Comment (marcoskwkm) [inline react/__mocks__/vtex.address-form.tsx]: I think you can just add `export` to the declarations (`export const components`, for instance) instead of adding this `module.exports` line.

File: react/components/ShippingInfo.tsx (hunk 2 above)
Comment (augustocb) [reply react/components/ShippingInfo.tsx:13]: Two points:
- `p` is a semantic tag for *paragraphs*
- `p` has a margin by default, that is not necessarily the expected one

**Review by marcoskwkm — CHANGES_REQUESTED** [review]
There are some other lint errors which you can check by running `yarn run eslint --ext tsx,ts .`

File: react/components/EstimateShipping.tsx (hunk 1 above)
Comment (jeffersontuc) [reply react/components/EstimateShipping.tsx:37]: I'm gonna try it.

File: react/__mocks__/vtex.address-form.tsx [hunk 8]
```diff
@@ -18,9 +18,9 @@ import AddressSummary from '@vtex/address-form/lib/AddressSummary'
 
 // Added react context for injectRules mock
 // eslint-disable-next-line
```
Comment (marcoskwkm) [inline react/__mocks__/vtex.address-form.tsx]: Do you still need these comments?

**Review by marcoskwkm — CHANGES_REQUESTED** [review]
`yarn run eslint --ext tsx,ts .` still reports some lint errors to me. Do you also them on your computer? I've listed some of them below.
```
/Users/marcos/vtex/shipping-calculator/react/__mocks__/vtex.address-form.tsx
  27:22  error  Component definition is missing display name  react/display-name
  33:26  error  Component definition is missing display name  react/display-name

/Users/marcos/vtex/shipping-calculator/react/__mocks__/vtex.shipping-estimate-translator.tsx
  2:3  error  Delete `··`  prettier/prettier
  3:2  error  Insert `⏎`   prettier/prettier

/Users/marcos/vtex/shipping-calculator/react/__mocks__/vtex.styleguide.tsx
  6:1  error  'module' is not defined  no-undef
```

**Review by marcoskwkm — APPROVED** [review]
lgtm
